src/test/java/LinkedIteratorTest.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class LinkedIteratorTest {
+
+    @Test
+    void hasNext_emptyLinked_returnsFalse() {
+        Node<Integer> head = null;
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        assertFalse(it.hasNext());
+    }
+
+    @Test
+    void hasNext_singleElementLinked_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementLinked_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_singleElementLinkedAfterHasNextAndNoNextCall_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.hasNext();
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementLinkedAfterHasNextAndNoNextCall_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_endOfSingleElementLinked_returnsFalse() {
+        Node<Integer> head = new Node<>(10);
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.next();
+        assertFalse(it.hasNext());
+    }
+
+    @Test
+    void hasNext_endOfSingleElementLinkedAfterHasNext_returnsFalse() {
+        Node<Integer> head = new Node<>(10);
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.next();
+        it.hasNext();
+        assertFalse(it.hasNext());
+    }
+
+    @Test
+    void hasNext_endOfMultiElementLinked_returnsFalse() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.next();
+        it.next();
```

Comment:
Same as comment on #218, I'd probably use a for loop to make it a little clearer how many times `next` is called here.